### ISSUE TITLE
Cache contract compilation error can cause node to crash

### ISSUE THREAD

**Issue by bowenwang1996** [near/nearcore#3655]
It appears that we cache errors in compiled contracts but do not expect errors when we deserialize the cached result (#3310). Therefore a node can crash if they fail to compile some contract and try to process a call to the contract:
```
Nov 25 21:09:26 testnet-canary0 neard[14488]: Nov 25 21:09:26.915 ERROR runtime: Cannot deserialize cached contract: FunctionCallError(CompilationEr
Nov 25 21:09:26 testnet-canary0 neard[14488]: thread 'main' panicked at 'Storage Error: StorageInconsistentState("Cache deserialization error")
Nov 25 21:09:26 testnet-canary0 neard[14488]:  Cause: Unknown', neard/src/runtime.rs:1073:21
```

**Comment by olonho** [comment]
See https://github.com/near/nearcore/pull/3656

**Comment by bowenwang1996** [comment]
Fixed in #3656
